services/community-backend/extensions/users-permissions/controllers/Auth.js

```diff
@@ -0,0 +1,304 @@
+    const user = await strapi
+      .query('user', 'users-permissions')
+      .findOne({ email: email.toLowerCase() });
+
+    // User not found.
+    if (!user) {
+      return ctx.badRequest(
+        null,
+        formatError({
+          id: 'Auth.form.error.user.not-exist',
+          message: 'This email does not exist.',
+        })
+      );
+    }
+
+    // Generate random token.
+    const resetPasswordToken = crypto.randomBytes(64).toString('hex');
+    const settings = await pluginStore.get({ key: 'email' }).then(storeEmail => {
+      try {
+        return storeEmail['reset_password'].options;
+      } catch (error) {
+        return {error: error};
+      }
+    });
+
+    const advanced = await pluginStore.get({
+      key: 'advanced',
+    });
+
+    const userInfo = sanitizeEntity(user, {
+      model: strapi.query('user', 'users-permissions').model,
+    });
+
+    settings.message = await strapi.plugins['users-permissions'].services.userspermissions.template(
+      settings.message,
+      {
+        URL: advanced.email_reset_password,
+        USER: userInfo,
+        TOKEN: resetPasswordToken,
+      }
+    );
+
+    settings.object = await strapi.plugins['users-permissions'].services.userspermissions.template(
+      settings.object,
+      {
+        USER: userInfo,
+      }
+    );
+
+    try {
+      // Send an email to the user.
+      await strapi.plugins['email'].services.email.send({
+        to: user.email,
+        from:
+          settings.from.email || settings.from.name
+            ? `${settings.from.name} <${settings.from.email}>`
+            : undefined,
+        replyTo: settings.response_email,
+        subject: settings.object,
+        text: settings.message,
+        html: settings.message,
+      });
+    } catch (err) {
+      return ctx.badRequest(null, err);
+    }
+
+    // Update the user.
+    await strapi.query('user', 'users-permissions').update({ id: user.id }, { resetPasswordToken });
+
+    ctx.send({ ok: true });
+  },
+
+
+    async register(ctx) {
+        const pluginStore = await strapi.store({
+          environment: '',
+          type: 'plugin',
+          name: 'users-permissions',
+        });
+    
+        const settings = await pluginStore.get({
+          key: 'advanced',
+        });
+    
+        if (!settings.allow_register) {
+          return ctx.badRequest(
+            null,
+            formatError({
+              id: 'Auth.advanced.allow_register',
+              message: 'Register action is currently disabled.',
+            })
+          );
+        }
+    
+        const params = {
+          ..._.omit(ctx.request.body, ['kyc', 'points', 'confirmed', 'confirmationToken', 'resetPasswordToken']),
+          provider: 'local',
+        };
+
+        if(!params.skipRecaptcha) {
+          const gres = await axios.post(`https://www.google.com/recaptcha/api/siteverify?secret=6LdLJXAaAAAAALsfpslEdIBHpW-OYKPuUwFT2lbY&response=${params.token}`)
```

Comment:
Here too. Move secret to env var

Reply:
@VargaElod23 handled pulling config from ENV and I have added `Secret` to helm-chart.